### ISSUE TITLE
Attention mask, is True True?

### ISSUE THREAD

**Issue by TKassis** [lucidrains/x-transformers#164]
Superb repo. Thank you. Pytorch should just merge this entire repo into official Pytorch.

I have quick clarification about attention mask syntax. I'm a bit confused regarding when attention is applied and when it's not, it seems that has changed a few times in different places and in different versions. As of today, would the following apply attention where attn_mask is True or where it is False?

```python

from x_transformers import ContinuousTransformerWrapper, Encoder 

self.transformer = ContinuousTransformerWrapper(
    max_seq_len= 512 + 1,
    attn_layers=Encoder(
        dim=d_model,
        depth=depth,
        heads=heads,
        attn_flash=True
    ),

x = self.transformer(x, mask=attn_mask)

```
Would attn_mask need to be True or False?

Thanks,

**Comment by lucidrains** [comment]
@TKassis oh hey Timothy from Boston! glad you still find the repository valuable for your work

so i was so confused by this early in my journey into transformers that i vowed to always stick with one convention

my convention, across all my repositories, is always `True` for attend, `False` for do not

**Comment by lucidrains** [comment]
@TKassis i would say learning attention and transformer should be super easy except for masking. masking has got to be the most confusing part (full of gotchas)

**Comment by TKassis** [comment]
Awesome, thanks for clarifying!

**Comment by lucidrains** [comment]
@TKassis do you know about xformers?

**Comment by TKassis** [comment]
Yes, but your repo is alot more feature rich.

**Comment by lucidrains** [comment]
ah yeah, just checking!
